MoEngagePluginCards/DataTransfer/Sync/MoEngageCardSyncManager.swift

```diff
@@ -0,0 +1,95 @@
+//
+//  MoEngageCardSyncManager.swift
+//  MoEngagePluginCards
+//
+//  Created by Soumya Mahunt on 20/06/23.
+//
+
+import MoEngageCore
+import MoEngageCards
+
+protocol MoEngageCardSyncManagerProtocol {
+    func attachDelegate(_ delegate: MoEngageCardSyncDelegate)
+    func detachDelegate()
+
+    func sendUpdate(
+        forEventType eventType: MoEngageCardsSyncEventType,
+        andAppID appId: String,
+        withNewData data: MoEngageCardSyncCompleteData?
+    )
+}
+
+final class MoEngageCardSyncManager: MoEngageCardSyncManagerProtocol {
+    private let queue: MoEngageCardSyncManagerSynchronizer
+    private let dataManager: MoEngageCardSyncDataManagerProtocol
+    private var delegate: MoEngageCardSyncDelegate?
+    private var syncedOnce: Bool
+
+    init(
+        queue: MoEngageCardSyncManagerSynchronizer = MoEngageCoreHandler.globalQueue,
+        dataManager: MoEngageCardSyncDataManagerProtocol = MoEngageCardSyncDataManager(),
+        syncedOnce: Bool = false
+    ) {
+        self.queue = queue
+        self.dataManager = dataManager
+        self.syncedOnce = syncedOnce
+    }
+
+    func attachDelegate(_ delegate: MoEngageCardSyncDelegate) {
+        queue.async {
+            self.delegate = delegate
+            for item in self.dataManager.dequeueSyncUpdateItems() {
+                self.sendUpdate(
+                    forEventType: item.eventType,
+                    andAppID: item.appId,
+                    withNewData: item.data
+                )
+            }
+        }
+    }
+
+    func detachDelegate() {
+        queue.async {
+            self.delegate = nil
+        }
+    }
+
+    func sendUpdate(
+        forEventType eventType: MoEngageCardsSyncEventType,
+        andAppID appId: String,
+        withNewData data: MoEngageCardSyncCompleteData?
+    ) {
+        queue.async {
+            let metadata = MoEngageCardSyncCompleteMetaData(type: eventType, data: data)
+            if let delegate = self.delegate {
+                let dataForHybrid = MoEngagePluginCardsUtil.buildHybridPayload(
+                    forIdentifier: appId,
+                    containingData: metadata.encodeForHybrid(forceUpdate: !self.syncedOnce)
+                )
+                DispatchQueue.main.async {
+                    delegate.syncComplete(
+                        forEventType: eventType,
+                        withData: dataForHybrid
```

Comment:
@msoumya-engg-sdk u need to pass simple json to hybrid side as per the contracts. Not model objects

Reply:
@RakshithaAcharya let me know your thoughts: https://github.com/moengage/iOS-PluginBase/pull/21#discussion_r1239461049